src/net.cpp

```diff
@@ -751,6 +766,18 @@ int CNetMessage::readData(const char *pch, unsigned int nBytes)
     memcpy(&vRecv[nDataPos], pch, nCopy);
     nDataPos += nCopy;
 
+    if (complete())
+    {
+        // reject messages with invalid checksum
+        uint256 hash = Hash(vRecv.begin(), vRecv.begin() + hdr.nMessageSize);
+        unsigned int nChecksum = ReadLE32((unsigned char*)&hash);
+        if (nChecksum != hdr.nChecksum)
+        {
+            LogPrint("net" "Message error: invalid checksum nChecksum=%08x hdr.nChecksum=%08x\n", __func__, nChecksum, hdr.nChecksum);
```

Comment:
I believe that format specifiers do not correspond to the arguments here, I guess you meant to remove `__func__`?

Reply:
Yes you are right, updated accordingly.